tests/plugins/find_and_replace.test.ts

```diff
@@ -102,194 +269,789 @@ describe("basic search", () => {
+        matchIndex = model.getters.getCurrentSelectedMatchIndex();
+        expect(getActivePosition(model)).toBe("A2");
+        expect(matches).toHaveLength(2);
+        expect(matchIndex).toStrictEqual(1);
+        expect(matches[0]).toStrictEqual({
+          sheetId: "s1",
+          col: 0,
+          row: 0,
+          selected: false,
+        });
+        expect(matches[1]).toStrictEqual({
+          sheetId: "s1",
+          col: 0,
+          row: 1,
+          selected: true,
+        });
+      } else if (searchScope === "allSheets") {
+        expect(getActivePosition(model)).toBe("A1");
+        expect(matches).toHaveLength(3);
+        expect(matchIndex).toStrictEqual(0);
+        expect(matches[0]).toStrictEqual({
+          sheetId: "s1",
+          col: 0,
+          row: 0,
+          selected: true,
+        });
+        expect(matches[1]).toStrictEqual({
+          sheetId: "s1",
+          col: 0,
+          row: 1,
+          selected: false,
+        });
+        expect(matches[2]).toStrictEqual({
+          sheetId: "Sheet2",
+          col: 0,
+          row: 2,
+          selected: false,
+        });
 
-    model.dispatch("SELECT_SEARCH_NEXT_MATCH");
-    matches = model.getters.getSearchMatches();
-    matchIndex = model.getters.getCurrentSelectedMatchIndex();
-    expect(getActivePosition(model)).toBe("A2");
-    expect(matches).toHaveLength(3);
-    expect(matchIndex).toStrictEqual(1);
-    expect(matches[0]).toStrictEqual({ col: 0, row: 0, selected: false });
-    expect(matches[1]).toStrictEqual({ col: 0, row: 1, selected: true });
-    expect(matches[2]).toStrictEqual({ col: 0, row: 2, selected: false });
-  });
-  test("search will cycle with previous", () => {
-    model.dispatch("UPDATE_SEARCH", { toSearch: "1", searchOptions });
-    let matches = model.getters.getSearchMatches();
-    let matchIndex = model.getters.getCurrentSelectedMatchIndex();
-    expect(getActivePosition(model)).toBe("A1");
-    expect(matches).toHaveLength(3);
-    expect(matchIndex).toStrictEqual(0);
-    expect(matches[0]).toStrictEqual({ col: 0, row: 0, selected: true });
-    expect(matches[1]).toStrictEqual({ col: 0, row: 1, selected: false });
-    expect(matches[2]).toStrictEqual({ col: 0, row: 2, selected: false });
+        model.dispatch("SELECT_SEARCH_NEXT_MATCH");
+        matches = model.getters.getSearchMatches();
+        matchIndex = model.getters.getCurrentSelectedMatchIndex();
+        expect(getActivePosition(model)).toBe("A2");
+        expect(matches).toHaveLength(3);
+        expect(matchIndex).toStrictEqual(1);
+        expect(matches[0]).toStrictEqual({
+          sheetId: "s1",
+          col: 0,
+          row: 0,
+          selected: false,
+        });
+        expect(matches[1]).toStrictEqual({
+          sheetId: "s1",
+          col: 0,
+          row: 1,
+          selected: true,
+        });
+        expect(matches[2]).toStrictEqual({
+          sheetId: "Sheet2",
```

Comment:
sheetID should be `s2`, it's confusion that the format of the ids of sheet1 and 2 are different, it looks like a sheet Id and a sheet name